src/GSoft.Extensions.MediatR/GSoft.Extensions.MediatR.csproj

```diff
@@ -32,4 +36,25 @@
     <InternalsVisibleTo Include="GSoft.Extensions.MediatR.ApplicationInsights,PublicKey=002400000480000094000000060200000024000052534131000400000100010025301ce547647ab5ac9264ade0f9cdc0252796a257095add4791b0232c1def21bb9e0c87d218713f918565b23394362dbcb058e210c853a24ec33e6925ebedf654a0d65efb3828c855ff21eaaa67aeb9b24b81b8baff582a03df6ab04424c7e53cacbfe84d2765ce840389f900c55824d037d2c5b6b330ac0188a06ef6869dba" />
     <InternalsVisibleTo Include="GSoft.Extensions.MediatR.Tests,PublicKey=002400000480000094000000060200000024000052534131000400000100010025301ce547647ab5ac9264ade0f9cdc0252796a257095add4791b0232c1def21bb9e0c87d218713f918565b23394362dbcb058e210c853a24ec33e6925ebedf654a0d65efb3828c855ff21eaaa67aeb9b24b81b8baff582a03df6ab04424c7e53cacbfe84d2765ce840389f900c55824d037d2c5b6b330ac0188a06ef6869dba" />
   </ItemGroup>
+
+  <PropertyGroup>
+    <!--
+    Insert custom target to embed our Roslyn analyzers to the generated package (https://stackoverflow.com/q/40396161/825695).
+    The idea is to insert the analyzers DLL to the corresponding analyzers directory during packing.
+    Microsoft does something similar to this with several packages including the source generators in Microsoft.Extensions.Logging.Abstractions.
+    -->
+    <TargetsForTfmSpecificBuildOutput>$(TargetsForTfmSpecificBuildOutput);CopyProjectReferencesToPackage</TargetsForTfmSpecificBuildOutput>
+  </PropertyGroup>
+
+  <Target Name="CopyProjectReferencesToPackage" DependsOnTargets="BuildOnlySettings;ResolveReferences">
```

Comment:
I've done that before with my [UI testing library](https://github.com/asimmon/marionette/) that includes a source generator.

It's better than publishing a separate NuGet package with the analyzers only. We want the roslyn analyzers to be part of the library itself.